modules/querying/pages/query-optimizer/index.adoc

```diff
@@ -0,0 +1,34 @@
+= Query Optimizer
+:description: Overview of the query optimizer.
+
+GSQL has a built-in query optimizer that can optimize the execution plan of a query when you install the query.
+Currently, the optimizer can optimize the execution plan in the following ways:
+
+* <<_cost_based_optimization>>
+
+== What is the query optimizer?
```

Comment:
I think we can merge this sub-header into the main `Query Optimizer` header.

Reply:
Since there will be more types added in the future, i want different options to be at the top. It also helps to open with a relatively simple description, and the delve a bit deeper for a detailed description later when we have something as complex as this.